src/utils/COMETAdapter.cpp

```diff
@@ -0,0 +1,577 @@
+    map_oms2comet["no cleavage"] = 0;   
+
+    String enzyme_name = getStringOption_("enzyme");
+    Size enzyme_number = 1;
+    if (map_oms2comet.find(enzyme_name) != map_oms2comet.end())
+    {
+      enzyme_number = map_oms2comet.at(enzyme_name);
+    }
+    else
+    {
+      throw OpenMS::Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Error: Enzyme not supported. " + enzyme_name);
+    }
+
+    os << "search_enzyme_number = " << enzyme_number << "\n";                // choose from list at end of this params file
+    os << "num_enzyme_termini = " << getIntOption_("num_enzyme_termini") << "\n";                  // 1 (semi-digested), 2 (fully digested, default), 8 C-term unspecific , 9 N-term unspecific
+    os << "allowed_missed_cleavage = " << getIntOption_("allowed_missed_cleavages") << "\n";             // maximum value is 5; for enzyme search
+
+    // Up to 9 variable modifications are supported
+    // format:  <mass> <residues> <0=variable/else binary> <max_mods_per_peptide> <term_distance> <n/c-term> <required>
+    //     e.g. 79.966331 STY 0 3 -1 0 0
+    vector<String> variable_modifications_names = getStringList_("variable_modifications");
+    vector<ResidueModification> variable_modifications = getModifications_(variable_modifications_names);
+    if (variable_modifications.size() > 9)
+    {
+      throw OpenMS::Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Error: Comet only supports 9 variable modifications. " + String(variable_modifications.size()) + " provided.");
+    }
+
+    Size var_mod_index = 1;
+
+    // write out user specified modifications
+    for (; var_mod_index <= variable_modifications.size(); ++var_mod_index)
+    {
+      const ResidueModification mod = variable_modifications[var_mod_index];
+      double mass = mod.getDiffMonoMass();
+      String residues = mod.getOrigin();  // TODO: check if origin contains C-term string or similar. Should not be passed to commet as residue string
+      String variable = "0";
+      String max_mods_per_peptide = "3";
+      String term_distance = "-1";
+      String nc_term = "0";
+
+      if (mod.getTermSpecificity() == ResidueModification::C_TERM) 
+      {
+        term_distance = 0;
+        nc_term = "3";
+      } 
+      else if (mod.getTermSpecificity() == ResidueModification::N_TERM)
+      {
+        term_distance = 0;
+        nc_term = "2";
+      } 
+      else if (mod.getTermSpecificity() == ResidueModification::PROTEIN_N_TERM)
+      {
+        term_distance = 0;
+        nc_term = "0";
+      }
+      else if (mod.getTermSpecificity() == ResidueModification::PROTEIN_C_TERM)
+      {
+        term_distance = 0;
+        nc_term = "1";
+      } 
+      String required = "0";
+
+      os << "variable_mod0" << var_mod_index << " = " << mass << " " << residues << " " << variable << " " << max_mods_per_peptide << " " << term_distance << " " << nc_term << " " << required << "\n"; 
```

Comment:
not sure what needs to happen here, depends on Comet nomenclature.
Either leave as is or ` << (var_mod_index+1) <<`